lib/new_relic/json.ex

```diff
@@ -0,0 +1,15 @@
+defmodule NewRelic.JSON do
+  @moduledoc false
+
+  if Code.ensure_loaded?(JSON) do
+    defdelegate decode(data), to: JSON
+    defdelegate decode!(data), to: JSON
+    defdelegate encode!(data), to: JSON
+  end
+
+  if Code.ensure_loaded?(Jason) do
+    defdelegate decode(data), to: Jason
+    defdelegate decode!(data), to: Jason
+    defdelegate encode!(data), to: Jason
+  end
```

Comment:
I think there will be many situations where both modules are loaded. That means we'd execute both of these blocks and call `defdelegate` twice...

It will be a long time before `Jason` actually drops off as a dependency, if ever. If it's a dependency for anything in an app, it'll be there.

* We should use a `cond` here to prioritize defining the functions
* We should not reference the module directly with defdelegate, or it'll warn about missing modules
* We should have a fallback that prompts the user

```suggestion
  cond do
    Code.ensure_loaded?(JSON) ->
      def decode(data), do: apply(JSON, :decode, [data])
      def decode!(data), do: apply(JSON, :decode!, [data])
      def encode!(data), do: apply(JSON, :encode!, [data])

    Code.ensure_loaded?(Jason) ->
      def decode(data), do: apply(Jason, :decode, [data])
      def decode!(data), do: apply(Jason, :decode!, [data])
      def encode!(data), do: apply(Jason, :encode!, [data])

    true ->
      raise "[:new_relic_agent] No JSON library found, please add :jason as a dependency"
  end

```